src/main/webapp/scripts/sessioncache.test.js

```diff
@@ -85,8 +72,14 @@ test('starting up after stopping', (done) => {
 
 test('retrieving info after starting immediately', async () => {
   fetch.mockResponse(JSON.stringify(expectedResult));
-  const cache = new SessionCache(testParams, 1000);
+  const cache = new SessionCache(testSessionRequest, 1000);
   cache.start();
   await expect(cache.getSession()).
       resolves.toEqual(expectedResult);
 });
+
+async function testSessionRequest() {
+  const response =
+      await fetch('https://api.exchangeratesapi.io/latest?base=USD');
+  return await response.json();
```

Comment:
Couldnt you just do:

const obj = await response.json();
return obj;

This way you dont have to have async in the function header? Does that work?

Reply:
nope, if you use await the function has to be async